extension/observer/endpoints.go

```diff
@@ -44,10 +44,10 @@ type Pod struct {
 
 // Port is an endpoint that has a target as well as a port.
 type Port struct {
-	Name     string
-	Pod      Pod
-	Port     uint16
-	Protocol Protocol
+	Name              string
+	Pod               Pod
+	Port              uint16
+	TransportProtocol TransportProtocol
```

Comment:
We call this "Transport" in a bunch of configs.

Reply:
Thanks for pointing that out. I was planning adding a `network_protocol` variable to endpoints (see https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/432#discussion_r456608431 for details). But a cleaner approach seems to be to have a variable called `transport` that would take `tcp`, `tcp4`, `tcp6` and `udp`, `udp4`, `udp6`.